### PR TITLE
(PUP-1795) Warn when numeric auto convert takes place

### PR DESCRIPTION
**Pull request by hlindberg**
See commit message

### CODE REVIEW

File: lib/puppet/pops/issues.rb [hunk 1]
```diff
@@ -506,6 +506,10 @@ def self.hard_issue(issue_code, *args, &block)
     "The value '#{value}' cannot be converted to Numeric."
   end
 
+  NUMERIC_COERCION = issue :NUMERIC_COERCION, :before, :after do
+    "The string '#{before}' was automatically coerced to the numerical value #{after}"
```
Comment (Magisus) [inline lib/puppet/pops/issues.rb]: Hmm... So in master, all of these errors have been externalized. How do we want to handle making sure that new errors also get externalized? Is that something that should happen post-mergeup? And if so, how are we going to remember to do it?

**Comment by thallgren** [comment]
Closing and reopening to make Travis choose correct base branch.

File: lib/puppet/pops/issues.rb (hunk 1 above)
Comment (hlindberg) [inline lib/puppet/pops/issues.rb]: A very good question

**Comment by hlindberg** [comment]
Note that the string in issues.rb must be externalized when merging up to master

**Comment by hlindberg** [comment]
travis build was failed after 5s - restarted build to see if that helps

**Comment by thallgren** [comment]
The travis build is back to thinking that this PR should be based on master. Not sure why that happens. I'm closing and reopening again to see if that has an effect (it worked yesterday).

**Pull request merged**